src/main/java/org/folio/event/handler/ItemCreateAsyncRecordHandler.java

```diff
@@ -96,4 +122,101 @@ private void updatePieceFields(List<Piece> pieces, String holdingId, String tena
+          var piecesFuture = pieceService.getPiecesByPoLineId(poLine.getId(), conn)
+            .map(allPieces -> Pair.of(poLine, allPieces));
+          poLinePiecePairsFutures.add(piecesFuture);
+        });
+        return collectResultsOnSuccess(poLinePiecePairsFutures);
+      })
+      .compose(poLinePiecePairs -> updatePoLines(poLinePiecePairs, itemObject, centralTenantId, headers, conn))
+      .mapEmpty();
+  }
+
+  // Update locations and searchLocationIds for each POL
+  private Future<List<PoLine>> updatePoLines(List<Pair<PoLine, List<Piece>>> poLinePiecePairs, JsonObject itemObject,
+                                             String centralTenantId, Map<String, String> headers, Conn conn) {
+    var updatedPoLines = new ArrayList<PoLine>();
+    log.info("updatePoLines:: Updating '{}' POL(s) for item: '{}' in centralTenant: '{}'",
+      poLinePiecePairs.size(), itemObject.getString(ID.getValue()), centralTenantId);
+    poLinePiecePairs.forEach(poLineListPair -> {
+      var poLine = poLineListPair.getLeft();
+      var pieces = poLineListPair.getRight();
+      var locationsUpdated = updatePoLineLocations(pieces, poLine);
+      var searchLocationIdsUpdated = updatePoLineSearchLocationIds(itemObject, poLine);
+      if (Boolean.TRUE.equals(locationsUpdated) || Boolean.TRUE.equals(searchLocationIdsUpdated)) {
+        updatedPoLines.add(poLine);
+      }
+    });
+    if (CollectionUtils.isEmpty(updatedPoLines)) {
+      log.info("updatePoLines:: No POLs were changed to update for item: '{}'", itemObject.getString(ID.getValue()));
+      return Future.succeededFuture(List.of());
+    }
+    return poLinesService.updatePoLines(updatedPoLines, conn, centralTenantId)
+      .compose(v -> auditOutboxService.saveOrderLinesOutboxLogs(conn, updatedPoLines, OrderLineAuditEvent.Action.EDIT, headers))
+      .mapEmpty();
+  }
+
+  private boolean updatePoLineLocations(List<Piece> pieces, PoLine poLine) {
+    var locations = new ArrayList<Location>();
+    var piecesByTenantIdGrouped = pieces.stream().filter(Objects::nonNull)
+      .filter(piece -> Objects.nonNull(piece.getReceivingTenantId()))
+      .collect(Collectors.groupingBy(Piece::getReceivingTenantId, Collectors.toList()));
+    piecesByTenantIdGrouped.forEach((tenantId, piecesByTenant) -> {
+      var piecesByHoldingIdGrouped = piecesByTenant.stream().filter(Objects::nonNull)
+        .filter(piece -> Objects.nonNull(piece.getHoldingId()))
+        .collect(Collectors.groupingBy(Piece::getHoldingId, Collectors.toList()));
+      piecesByHoldingIdGrouped.forEach((holdingId, piecesByHoldings) -> {
+        var piecesByFormat = piecesByHoldings.stream() .filter(Objects::nonNull)
+          .filter(piece -> Objects.nonNull(piece.getFormat()))
+          .collect(Collectors.groupingBy(Piece::getFormat, Collectors.toList()));
+        var location = new Location().withTenantId(tenantId)
+          .withHoldingId(holdingId)
+          .withQuantity(piecesByHoldings.size())
```

Comment:
Possible case:
1 piece which its locationId is exists and holdingid is null is updated ownership from university to college and trying to update location with holdingSize that may not be correct.

Reply:
> Possible case: 1 piece which its locationId is exists and holdingid is null is updated ownership from university to college and trying to update location with holdingSize that may not be correct.

We've discussed this with Serhii, seems like only `holdingId` will be present, otherwise this code will purge all `locationId`-based locations